lib/button-group.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Button } from './button';
+import { onArrowKeys } from './keyboard-navigation';
+import { code, CodeExample, PropsDefinition, Prop } from './story-utils';
+import { Icon } from './icon';
+
+const ButtonWrap = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+export const ButtonSegment = styled(Button)`
+  flex: 0 0 auto;
+  border-radius: 0;
+  border-right-style: none;
+  &[aria-checked='true'] {
+    color: var(--colors-selected-text);
+    background-color: var(--colors-selected-background);
```

Comment:
Looks like the current colour scheme doesn't match the community default or ours. Could we change it to match?
![Screenshot 2019-08-14 at 11 13 16 AM](https://user-images.githubusercontent.com/9853740/63032939-ca452380-be84-11e9-9611-7df91acb4813.png)
![Screenshot 2019-08-14 at 11 15 26 AM](https://user-images.githubusercontent.com/9853740/63032961-d5984f00-be84-11e9-8cc9-ca5a1672262a.png)
@ehmorris ok with dropping the bold weight to match community?

Reply:
I was just about to comment on this. I think that using the selected blue might ultimately be better for usability since gray button states can look disabled.

Looking at the editor, we use 4 different button styles for tabbed buttons. In this screenshot, the leftmost button is the current tab, the middle is `:active`, the rightmost button is `:hover`, and none of the buttons are shown in the default state.
<img width="217" alt="Screen Shot 2019-08-14 at 11 23 50 AM" src="https://user-images.githubusercontent.com/654275/63033671-02993180-be86-11e9-9451-2451677b7766.png">

We shouldn't replicate this because the relationship these visual states have to each other doesn't really make sense. It's just a jumble of gray.

I'm not crazy about putting dark blue inside a dark gray border, but I think for now this is a better solution. It also looks much better in dark mode.

I will say that I definitely don't want to see us using this "active tab" color for other cursor states, like hover:

<img width="369" alt="Screen Shot 2019-08-14 at 11 36 08 AM" src="https://user-images.githubusercontent.com/654275/63034722-cebf0b80-be87-11e9-9024-4146363ac0bb.png">

^ When it comes time to componentize that component, we'll similarly have to make a change.

-----

Also of note, looking at the shared components buttons, we only have a normal, hover, and focus state, and the focus state is just the default browser outline. We don't have `:active` (e.g. mousedown) states on any of the buttons.

I think we should add an `:active` state for all buttons, including these, which darkens the background, border, and text colors (or just the background for buttons where the border and text are already black.)

Reply:
How would you feel about using the secondary background color for hover, and the dark blue for both selected and `:active`?

Reply:
Just tested this out and it seems alight. The black border thing isn't ideal but it's ok. One thing that would be nice to see is applying a white color to the arrow icons on active.

![Screencast2](https://user-images.githubusercontent.com/654275/63178474-1d9bab00-c018-11e9-8947-64db2400d23b.gif)
![Screencast](https://user-images.githubusercontent.com/654275/63178475-1d9bab00-c018-11e9-9293-14cee5d9585b.gif)
![Screencast3](https://user-images.githubusercontent.com/654275/63178575-563b8480-c018-11e9-839d-02b0db14d50e.gif)

Reply:
Also maybe we should change the z-index or something to pop the active button to the top so we can fix the weird focus ring issues?
![Screenshot 2019-08-16 at 3 38 54 PM](https://user-images.githubusercontent.com/9853740/63193664-0e2e5900-c03c-11e9-9920-20f9ef93d56e.png)

Reply:
updated